src/track/trackmetadatataglib.cpp

```diff
@@ -1026,6 +1026,19 @@ void readTrackMetadataFromID3v2Tag(TrackMetadata* pTrackMetadata,
     const TagLib::ID3v2::FrameList bpmFrame(tag.frameListMap()["TBPM"]);
     if (!bpmFrame.isEmpty()) {
         parseBpm(pTrackMetadata, toQStringFirstNotEmpty(bpmFrame));
+        long bpmValue = pTrackMetadata->getBpm().getValue();
+        //Some software uses (or used) to write decimated values without comma,
+        //so the number reads as 1352 or 14525 when it is 135.2 or 145.25
+        if (bpmValue >= Bpm::kValueMinForTwoDecimals) {
+            qWarning() << " Changing BPM on " << pTrackMetadata->getArtist() << " - " << 
+                pTrackMetadata->getTitle() << " from " << bpmValue << " to " << bpmValue/100.0;
+            bpmValue /= 100.0;
+        } else if (bpmValue >= Bpm::kValueMinForOneDecimal) {
+            qWarning() << " Changing BPM on " << pTrackMetadata->getArtist() << " - " << 
+                pTrackMetadata->getTitle() << " from " << bpmValue << " to " << bpmValue/10.0;
+            bpmValue /= 10.0;
+        }
+        pTrackMetadata->getBpm().setValue(bpmValue);
```

Comment:
Ok, i'll rewrite it as a loop (i was considering it, but we didn't had a max value defined (in fact, there's one defined in the test case, but not in the bpm object)).

As for the tag... The reason I wanted to change it is because it affected Mixxx usage (from bad ordering in the library to showing an unrecognizable beatgrid). 
So, what I had to do inevitably was to correct the BPM value by hand.
And If I need to change the value, I am at the same case than with the automatic change if ID3 writing comes back.
In this regard, I consider that worrying about writing a different value than the original woudn't matter (More so with uklotzde comment below).